### PR TITLE
COMPAT: Replace ContentType if copyObject with REPLACE

### LINKED ISSUE #340: COMPAT: should replace ContentType if copyObject with REPLACE
Currently we do not replace `ContentType` even if the metadata directive header REPLACE is specified for `x-amz-metadata-directive` during a `copyObject` request. However, AWS replaces this header if REPLACE is specified and a new `ContentType` header is provided in the `copyObject` request. If a new header is not provided, it keeps the original ContentType header.

On **ceph/s3-tests**, these changes fix:
`ERROR: s3tests.functional.test_s3.test_object_copy_replacing_metadata`

### PR DESCRIPTION
**Pull request by electrachong**
FIXES #340

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine if you would like

**Comment by electrachong** [comment]
build failed in previous test but this rebuild without cache succeeds: http://ci.ironmann.io/gh/scality/Integration/5841

File: tests/functional/aws-node-sdk/test/object/objectCopy.js [hunk 1]
```diff
@@ -214,6 +214,38 @@ describe('Object Copy', () => {
...
+            'included as a metadata directive header, and new ContentType is ' +
+            'provided', done => {
+            s3.copyObject({ Bucket: destBucketName, Key: destObjName,
+                CopySource: `${sourceBucketName}/${sourceObjName}`,
+                MetadataDirective: 'REPLACE',
+                ContentType: 'image',
+            }, () => {
+                s3.getObject({ Bucket: destBucketName,
+                Key: destObjName }, (err, res) => {
+                    assert.strictEqual(res.ContentType, 'image');
```
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/object/objectCopy.js]: check `if (err) ... done(err)`

File: tests/functional/aws-node-sdk/test/object/objectCopy.js [hunk 2]
```diff
@@ -214,6 +214,38 @@ describe('Object Copy', () => {
...
+        it('should copy an object and keep ContentType if replace ' +
+            'included as a metadata directive header, but no new ContentType ' +
+            'is provided', done => {
+            s3.copyObject({ Bucket: destBucketName, Key: destObjName,
+            CopySource: `${sourceBucketName}/${sourceObjName}`,
+            MetadataDirective: 'REPLACE',
+        }, () => {
+                s3.getObject({ Bucket: destBucketName,
+                    Key: destObjName }, (err, res) => {
+                    assert.strictEqual(res.ContentType,
```
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/object/objectCopy.js]: Same here: check if err

File: tests/functional/aws-node-sdk/test/object/objectCopy.js (hunk 1 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/objectCopy.js]: as you like, it's done :)

File: tests/functional/aws-node-sdk/test/object/objectCopy.js (hunk 2 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/objectCopy.js]: done

File: tests/functional/aws-node-sdk/test/object/objectCopy.js (hunk 1 above)
Comment (electrachong) [inline tests/functional/aws-node-sdk/test/object/objectCopy.js]: ok, done again :P

File: lib/api/objectCopy.js [hunk 3]
```diff
@@ -80,6 +80,12 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
         lastModifiedDate: new Date().toJSON(),
     };
 
+    // If whichMetadata === 'REPLACE' but contentType is undefined in response
```
Comment (LaurenSpiegel) [inline lib/api/objectCopy.js]: I think you mean 'copy request headers' instead of 'response'?

Comment (electrachong) [inline lib/api/objectCopy.js]: yeah, that's what I meant :P thanks

**Pull request merged**